SplunkRumWorkspace/SplunkRum/SplunkRum/ScreenFrames.swift

```diff
@@ -0,0 +1,104 @@
+//
+/*
+Copyright 2021 Splunk Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import QuartzCore
+import UIKit
+
+class ScreenFrames: NSObject {
+
+    private var displayLink: CADisplayLink?
+
+    private var slowFrameThreshold: CFTimeInterval = 1.0 / 59.0
+    private var frozenFrameThreshold: CFTimeInterval = 700.0 / 1000.0
+
+    private var currentIteration: Int = 0
+    private var startedTime: CFTimeInterval = CACurrentMediaTime()
+
+    private var slowCount: Int = 0
+    private var frozenCount: Int = 0
+    private var isFirstIteration: Bool = true
+    private var previousTimestamp: CFAbsoluteTime = CACurrentMediaTime()
+
+    func startTracking() {
+
+        stopTracking() /// make sure to stop a previous running display link
+        let displayLink = CADisplayLink(target: self, selector: #selector(displayLinkCallback))
+        displayLink.add(to: .main, forMode: .common)
+        self.displayLink = displayLink
+    }
+
+    func stopTracking() {
+
+        displayLink?.invalidate()
+        displayLink = nil
+    }
+
+    @objc func displayLinkCallback(_ displayLink: CADisplayLink) {
+
+         let state = UIApplication.shared.applicationState
```

Comment:
Instead of checking for the state here and creating a new CADisplayLink on every VC appearance, would it work to use `applicationDidEnterBackground` and `applicationDidBecomeActive` callbacks instead?